frontend/tests/components/applyForm/ApplyForm.test.tsx

```diff
@@ -124,4 +124,20 @@ describe("ApplyForm", () => {
       expect.any(FormData),
     );
   });
+  it("errors when form data is bad", () => {
```

Comment:
trying to figure out the error case here - it looks like it's the empty object `formSchema` prop that's causing the error rather than `savedFormData`. If that's true can that be more clear in the name of the test? Related, but separate question for later - is there also a possible error case where the formSchema is valid but the savedFormData is bad?

Reply:
I updated the test to be more explicit and contain cases of empty objects and ones that are not JSON Schema compliant.

> Related, but separate question for later - is there also a possible error case where the formSchema is valid but the savedFormData is bad?

This should fail gracefully. The form schema is the source of truth. Down the line we might want to make the form fail if the data does not conform to the JSON schema.

I included a test to verify the above.